### PR TITLE
Fix MeshNetworks example documentation

### PR DESCRIPTION
**Pull request by cainelli**
The example for MeshNetworks in the [documentation page](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshNetworks) doesn't seem to be correct. The generated pb.go file looks like this:

https://github.com/istio/api/blob/11df83b4d252c9eee246fc26aaa5d4fc2e0acd78/mesh/v1alpha1/network.pb.go#L327-L331

### CODE REVIEW

File: mesh/v1alpha1/network.proto [hunk 1]
```diff
@@ -102,7 +102,7 @@ message Network {
 // ```yaml
 // networks:
 //   network1:
-//   - endpoints:
+//     endpoints:
```
Comment (howardjohn) [inline mesh/v1alpha1/network.proto:105]:
```
type Network struct {
	// The list of endpoints in the network (obtained through the
	// constituent service registries or from CIDR ranges). All endpoints in
	// the network are directly accessible to one another.
	Endpoints []*Network_NetworkEndpoints `protobuf:"bytes,2,rep,name=endpoints,proto3" json:"endpoints,omitempty"`
```

seems like endpoints is a repeated fiield?

Comment (cainelli) [reply mesh/v1alpha1/network.proto:105]: Yes, this seems to be allowed (at least in the configuration parsing) in the configuration:

```yaml
networks:
   network1:
     endpoints:
     - fromRegistry: registry1 #must match kubeconfig name in Kubernetes secret
     - fromCidr: 192.168.100.0/22 #a VM network for example
```

but not this:

```yaml
networks:
   network1:
   - endpoints:
     - fromRegistry: registry1 #must match kubeconfig name in Kubernetes secret
     - fromCidr: 192.168.100.0/22 #a VM network for example
```

Comment (howardjohn) [reply mesh/v1alpha1/network.proto:105]: ah right.. yaml tricks me every time :slightly_smiling_face:

**Review by smawson — APPROVED** [review]
YAML is hard!

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**